test/plugin/pom.xml

```diff
@@ -99,6 +103,151 @@
                     <encoding>${project.build.sourceEncoding}</encoding>
                 </configuration>
             </plugin>
+            <plugin>
+                <artifactId>maven-checkstyle-plugin</artifactId>
+                <version>${maven-checkstyle-plugin.version}</version>
+                <configuration>
+                    <configLocation>${maven.multiModuleProjectDirectory}/apm-checkstyle/checkStyle.xml</configLocation>
+                    <headerLocation>${maven.multiModuleProjectDirectory}/apm-checkstyle/CHECKSTYLE_HEAD</headerLocation>
+                    <encoding>UTF-8</encoding>
+                    <consoleOutput>true</consoleOutput>
+                    <includeTestSourceDirectory>true</includeTestSourceDirectory>
+                    <failOnViolation>${checkstyle.fails.on.error}</failOnViolation>
+                    <sourceDirectories>
+                        <sourceDirectory>${project.build.sourceDirectory}</sourceDirectory>
```

Comment:
Due to https://github.com/apache/skywalking/blob/753c73ab8ba68717d67dfa5306a7d224a6f26805/test/plugin/generator.sh#L79-L84, could we give rid of this?

Reply:
Because `scenarios` is not a maven project, if only configure `<sourceDirectory>${project.build.sourceDirectory}</sourceDirectory>`, the maven project in `scenarios` can't  be checked.

Reply:
I know, I mean you are using this script to generate these configurations, right? But the hard-code configurations are still there, https://github.com/apache/skywalking/blob/753c73ab8b/test/plugin/pom.xml#L116-L222

Do I misunderstand something?

Reply:
The `generator.sh` is to generate **NEW** scenario project, and it will add a new line of that project to https://github.com/apache/skywalking/blob/753c73ab8b/test/plugin/pom.xml#L116-L222

Reply:
Why limit it to new only? Why don't we do that before running the tests and check the code style first?

Reply:
It will be ok to write this script into `run.sh`. If we do so, it will ensure that the code style is checked before running.

Reply:
Yes, then this part could be removed and don't require the new contributor to add.

Reply:
There are some problems that should be considered, if we write the scripts into `run.sh`:
1. We execute scenarios in a different order and the content of pom.xml may be different, which breaks idempotence and may cause conflicts when merging.
2. Some scenarios may need to be handled separately to check the code style. For example `jdk14-with-gson-scenario` cannot be checked by adding its name in `plugin/pom.xml` because of different jdk version.

Reply:
> The changed pom.xml wouldn't be submitted or merged. It is updated in the CI process only.

Add the scenario name into `pom.xml` before run, and delete the added content after execution. Is it like this? 



> Could you be more clear about what does this mean? Why does code style relate to JDK version?

When I add ` <sourceDirectory>scenarios/jdk14-with-gson-scenario</sourceDirectory>` into `plugin/pom.xml`, We can see the following verification results：
`Failed to execute goal org.apache.maven.plugins:maven-checkstyle-plugin:3.1.0:check (validate) on project apache-skywalking-test-plugin: Failed during checkstyle configuration
`
So, I guess the reason may be caused by different jdk versions and I handle it separately. If my understanding is wrong or you have a better solution please let me know

Reply:
With your description, how about adding the check style plugin into every scenario's pom.xml, rather than governing the parent pom.xml?

Reply:
The generator.sh you changed clearly has the chance to add codestyle check when setting a new plugin test scenario, right?

Reply:
Yes,  it will add `<sourceDirectory>scenario_name<\/sourceDirectory>` into pom.xml . It works well when I test on my pc.

Reply:
I don't mean this, `Yes, it will add <sourceDirectory>scenario_name<\/sourceDirectory> into pom.xml`.

I prefer you consider to add the plugin configuration into file like this one, https://github.com/apache/skywalking/blob/9c48d0686a/test/plugin/scenarios/activemq-scenario/pom.xml

Reply:
ok，I got it. I will have a try!